### PR TITLE
provide simpletype to ejs templates for reuse at nested props

### LINKED ISSUE #69: known values of enum empty
## What did you do

I defined `enum` within a nested schema and the values get lost somehow.

## What did you expect to happen

I expect that all values show up the **Known Values** table.

## What happened

None are shown.

**Note:** I run all my schemas through [ajv](https://ajv.js.org/) to validate them - which are all fine.

## What's your environment

* Operating System: Mac OS X El Capitan
* node.js version: `v9.4.0`

## Do you have example files:

Sure, I even have a [PR](https://github.com/axa-ch/aletheia/pull/93) of a public repo.

For this schema
```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "$id": "http://www.axa.ch/schemas/components/Typo.json",

  "type": "object",
  "title": "Typo",
  "description": "Renders a Typo.",

  "additionalProperties": false,
  "required": ["component"],
  "properties": {
    "component": {
      "allOf": [
        { "$ref": "http://www.axa.ch/schemas/types/component.json#" },
        { "enum": ["Typo"] }
      ]
    },

    "properties": {
      "type": "object",
      "title": "Properties",
      "description": "Properties of Typo.",

      "additionalProperties": false,
      "properties": {
        "value": {
          "type": "string",
          "title": "Value",
          "description": "The text to be displayed."
        }
      }
    },

    "style": {
      "type": "object",
      "title": "Style",
      "description": "Style of Typo.",

      "additionalProperties": false,
      "properties": {
        "title": {
          "type": "boolean",
          "title": "Title",
          "description": "Whether or not the text is a heading.",
          "default": false
        },
        "type": {
          "type": "string",
          "title": "Type",
          "description": "The type of heading to be used for display.",
          "enum": ["main", "page", "slice", "small-module"],
          "default": "main"
        },
        "publico": {
          "type": "boolean",
          "title": "Publico",
          "description": "Whether or not to use the Publico Headline font in favour of Source Sans Pro",
          "default": false
        },
        "size": {
          "type": "string",
          "title": "Size",
          "description": "Size of Typo.",
          "enum": ["semibold", "bold", "normal"],
          "default": "normal"
        },
        "align": {
          "type": "string",
          "title": "Align",
          "description": "Set CSS text-align accordingly.",
          "enum": ["left", "right", "center"],
          "default": "left"
        },
        "longer": {
          "type": "boolean",
          "title": "Longer",
          "description": "Whether or not longer style is used for display.",
          "default": false
        },
        "secondary": {
          "type": "boolean",
          "title": "Secondary",
          "description": "Whether or not to display text according to AXA.ch style guide's secondary definition.",
          "default": false
        },
        "secondRow": {
          "type": "boolean",
          "title": "Second Row",
          "description": "Whether or not to use smaller margin between to show that two typos belong together.",
          "default": false
        },
        "color": {
          "type": "string",
          "title": "Color",
          "description": "Color of Typo.",
          "enum": ["red", "blue", "black"],
          "default": "black"
        }
      }
    }
  }
}
```

I'm getting following Markdown

```markdown

# Typo Schema

```
http://www.axa.ch/schemas/components/Typo.json
```

Renders a Typo.

| Abstract | Extensible | Status | Identifiable | Custom Properties | Additional Properties | Defined In |
|----------|------------|--------|--------------|-------------------|-----------------------|------------|
| Can be instantiated | No | Experimental | No | Forbidden | Forbidden | [components/Typo.json](components/Typo.json) |

# Typo Properties

| Property | Type | Required | Defined by |
|----------|------|----------|------------|
| [component](#component) | complex | **Required** | Typo (this schema) |
| [properties](#properties) | `object` | Optional | Typo (this schema) |
| [style](#style) | `object` | Optional | Typo (this schema) |

## component


`component`
* is **required**
* type: complex
* defined in this schema

### component Type


**All** of the following *requirements* need to be fulfilled.


#### Requirement 1


* []() – `http://www.axa.ch/schemas/types/component.json#`


#### Requirement 2







## properties
### Properties

Properties of Typo.

`properties`
* is optional
* type: `object`
* defined in this schema

### properties Type


`object` with following properties:


| Property | Type | Required
|----------|------|----------|
| `value`| string | Optional | 



#### value
##### Value

The text to be displayed.

`value`
* is optional
* type: `string`

##### value Type


`string`











## style
### Style

Style of Typo.

`style`
* is optional
* type: `object`
* defined in this schema

### style Type


`object` with following properties:


| Property | Type | Required
|----------|------|----------|
| `align`| string | Optional | 
| `color`| string | Optional | 
| `longer`| boolean | Optional | 
| `publico`| boolean | Optional | 
| `secondRow`| boolean | Optional | 
| `secondary`| boolean | Optional | 
| `size`| string | Optional | 
| `title`| boolean | Optional | 
| `type`| string | Optional | 



#### align
##### Align

Set CSS text-align accordingly.

`align`
* is optional
* type: `string`

The value of this property **must** be equal to one of the [known values below](#style-known-values).

##### align Known Values
| Value | Description |
|-------|-------------|






#### color
##### Color

Color of Typo.

`color`
* is optional
* type: `string`

The value of this property **must** be equal to one of the [known values below](#style-known-values).

##### color Known Values
| Value | Description |
|-------|-------------|






#### longer
##### Longer

Whether or not longer style is used for display.

`longer`
* is optional
* type: `boolean`

##### longer Type


`boolean`







#### publico
##### Publico

Whether or not to use the Publico Headline font in favour of Source Sans Pro

`publico`
* is optional
* type: `boolean`

##### publico Type


`boolean`







#### secondRow
##### Second Row

Whether or not to use smaller margin between to show that two typos belong together.

`secondRow`
* is optional
* type: `boolean`

##### secondRow Type


`boolean`







#### secondary
##### Secondary

Whether or not to display text according to AXA.ch style guide's secondary definition.

`secondary`
* is optional
* type: `boolean`

##### secondary Type


`boolean`







#### size
##### Size

Size of Typo.

`size`
* is optional
* type: `string`

The value of this property **must** be equal to one of the [known values below](#style-known-values).

##### size Known Values
| Value | Description |
|-------|-------------|






#### title
##### Title

Whether or not the text is a heading.

`title`
* is optional
* type: `boolean`

##### title Type


`boolean`







#### type
##### Type

The type of heading to be used for display.

`type`
* is optional
* type: `string`

The value of this property **must** be equal to one of the [known values below](#style-known-values).

##### type Known Values
| Value | Description |
|-------|-------------|









```

### PR DESCRIPTION
**Pull request by AndyOGo**
Thanks for this nice doc generator.
This fixes #69 `enum`'s within nested properties don't get displayed.

### CODE REVIEW

**Comment by trieloff** [comment]
@AndyOGo thanks for the PR, I would ask for following addition:

- add a test that illustrates the issue (i.e. one that would have failed without the patch)
- make the tests pass

The change itself looks good so far.

**Comment by AndyOGo** [comment]
@trieloff 
I updated the enum example to include a nested object with an enum property.

**Comment by AndyOGo** [comment]
@trieloff 
@anjankaur 
Is my provided test sufficient for your needs?
I would be very happy to see this merged and released soon.

File: spec/examples/complex.schema.md [hunk 1]
```diff
@@ -150,7 +150,7 @@ This is not foo.
 
 `nonfoo`
 * is optional
-* type: `boolean`
+* type: `const`
```
Comment (anjankaur) [inline spec/examples/complex.schema.md]: @AndyOGo Why the change in type here?

Comment (AndyOGo) [reply spec/examples/complex.schema.md]: @anjankaur 
because it's schema defines it's boolean value by `const` here:
https://github.com/adobe/jsonschema2md/blob/4bf35ddacdd53a1e5d1112fb0f2a0f28e3568323/examples/schemas/abstract.schema.json#L21-L25

And the `simpletype` function checks for `const` prop with higher priority than the `type` prop, here:
https://github.com/adobe/jsonschema2md/blob/2cc177c681e4b63d0d2784b8bc2f373126ebb162/lib/markdownWriter.js#L94-L102
